### PR TITLE
feat: agent - eBPF Create maps based on functional configuration

### PR DESCRIPTION
**Pull request by yinjiping**
Memory usage of maps corresponding to eBPF capabilities in DeepFlow:

| Uprobe Trace [1]  | Continuous Profile | Pre-allocating Memory| Dwarf Unwinding [1] |Memory used (Bytes)|
 | ----------------------- | --------------------------| ----------------------|---|---|
 |  Disable |  Disable |Disable |Disable | 13,674,008|
| **Enable** | Disable  | Disable |Disable | 30,909,632|
 |  Disable  | **Enable**  | Disable |Disable | 31,731,736|
| Disable  | Disable  | **Enable** |Disable|59,557,832|
| Disable  | **Enable**|Disable  |**Enable [2]**|182,281,992|
| Disable  | **Enable**|**Enable**  |**Enable [2]**|228,165,816|
| **Enable**  | **Enable**|**Enable**  |**Enable [2]**|263,095,728|
 |  Disable  | **Enable**  | **Enable** |Disable |77,615,560 [3]|

Additional note: 
 - [1]:  Disabled by default.
 - [2]:  DWARF unwinding relies on Continuous Profile.
 - [3]:  Memory usage in the default configuration.
 
**Optimization Results**:
- When Uprobe Trace and DWARF Unwinding are disabled, this optimization reduces memory usage by 17,235,624MB compared to the previous version.
- If `Disable Pre-allocating Memory` is set, this optimization reduces memory usage by 45,883,824MB compared to the previous version.

In total, this optimization reduces memory usage by 63,119,448MB compared to the previous version.
 

### This PR is for:

- Agent

#### Affected branches
- main

#8028

### CODE REVIEW

File: server/agent_config/example.yaml [hunk 1]
```diff
@@ -1312,6 +1312,20 @@ static_config:
...
+    #   be traced, and the maximum number of entries in the associated BPF map will be set to 1 to
+    #   ensure minimal memory usage.
+    uprobe-golang-trace-enabled: false
+
+    # Enabled uprobe openssl library tracing
+    # Default: false
+    # Note: If this configuration option is set to false, the openssl library will not be traced,
+    #   and the maximum number of entries in the associated BPF map will be set to 1 to ensure
+    #   minimal memory usage.
+    uprobe-openssl-trace-enabled: false
```
Comment (yinjiping) [inline server/agent_config/example.yaml:1327]: 加了两项配置，uprobe-golang-trace-enabled 和 uprobe-openssl-trace-enabled。 
原因是：bpf map和ebpf的功能（包含 socket tracer， uprobe golang trace，uprobe opennssl trace，oncpu profile, offcpu profile, memory profile, dwarf unwinding）进行了关联。如果某项功能不启用，那么这个功能关联的maps将会设置为最少量内存（entry-max : 1）占用，实现是在ebpf加载之前进行判断，所以需要在加载之前明确告诉ebpf golang/openssl Trace 是开启还是关闭。

调用接口：
```
    pub fn set_uprobe_golang_enabled(enabled: bool) -> c_void;
    pub fn set_uprobe_openssl_enabled(enabled: bool) -> c_void;
```

在 running_socket_tracer() 之前调用。

Comment (yinjiping) [reply server/agent_config/example.yaml:1327]: @yuanchaoa 帮忙rust部分实现一下

File: server/agent_config/template.yaml [hunk 2]
```diff
@@ -2183,7 +2183,8 @@ inputs:
           #     and auto-tracing. go auto-tracing also dependent go-tracing-timeout.
           #   ch: |-
           #     Golang 程序 HTTP2/HTTPS 协议数据采集及零侵扰追踪特性的开启开关。
-          # upgrade_from: static_config.ebpf.uprobe-process-name-regexs.golang
+          # upgrade_from: static_config.ebpf.uprobe-golang-trace-enabled,
+          #               static_config.ebpf.uprobe-process-name-regexs.golang
```
Comment (sharang) [inline server/agent_config/template.yaml:2187]: 这里要弄成两行吗？

Comment (yinjiping) [reply server/agent_config/template.yaml]: 修改为1行了

**Review by rvql — APPROVED** [review]

**Pull request merged**